src/hooks/useCollections.ts

```diff
@@ -12,38 +14,69 @@ type StacCollectionsHook = {
 };
 
 function useCollections(): StacCollectionsHook {
-  const { stacApi, collections, setCollections } = useStacApiContext();
+  const { stacApi, setCollections } = useStacApiContext();
   const [state, setState] = useState<LoadingState>('IDLE');
-  const [error, setError] = useState<ApiError>();
 
-  const _getCollections = useCallback(() => {
-    if (stacApi) {
-      setState('LOADING');
+  const fetchCollections = async (): Promise<CollectionsResponse> => {
+    if (!stacApi) throw new Error('No STAC API configured');
+    const response: Response = await stacApi.getCollections();
+    if (!response.ok) {
+      let detail;
+      try {
+        detail = await response.json();
+      } catch {
+        detail = await response.text();
+      }
+
+      const err = Object.assign(new Error(response.statusText), {
+        status: response.status,
+        statusText: response.statusText,
+        detail,
+      });
+      throw err;
```

Comment:
Instead of extending the error object we could create a new error class, with the needed fields and a proper constructor. This would also give us cleaner types.